### PR TITLE
Add gradient output functionality in Neural Nets

### PR DESCRIPTION
**Pull request by sanuj**
@lisitsyn Have a look. Currently I'm printing the max and mean gradients after each iteration. Something like: https://gist.github.com/sanuj/4994d1d8feccdf069400

### CODE REVIEW

File: src/shogun/neuralnets/NeuralNetwork.cpp [hunk 1]
```diff
@@ -309,6 +309,18 @@ bool CNeuralNetwork::train_gradient_descent(SGMatrix<float64_t> inputs,
 
 			float64_t e = compute_gradients(inputs_batch, targets_batch, gradients);
 
+
+			for (int32_t k=0; k<m_num_layers; k++)
```
Comment (sanuj) [inline src/shogun/neuralnets/NeuralNetwork.cpp]: @lisitsyn This for loop can be made a private member function if you want as it is getting repeated in lbfgs also.

File: src/shogun/neuralnets/NeuralNetwork.cpp [hunk 2]
```diff
@@ -408,6 +420,19 @@ int CNeuralNetwork::lbfgs_progress(void* instance,
...
 	SG_SINFO("Epoch %i: Error = %f\n",k, fx);
+
+	CNeuralNetwork* network = static_cast<CNeuralNetwork*>(instance);
+	SGVector<float64_t> gradients((float64_t*)g, network->get_num_parameters(), false);
+	for (int32_t i=0; i<network->m_num_layers; i++)
+	{
+		SGVector<float64_t> layer_gradients = network->get_section(gradients, i);
+		if (layer_gradients.vlen > 0)
+		{
+			SG_SINFO("Layer %i, Max Gradient: %g, Mean Gradient: %g.\n", i,
```
Comment (lisitsyn) [inline src/shogun/neuralnets/NeuralNetwork.cpp]: For convenience, can we output the layer name as well?

Comment (sanuj) [inline src/shogun/neuralnets/NeuralNetwork.cpp]: Alright, i'll do it.

**Comment by sanuj** [comment]
@lisitsyn Done. Have a look.

**Comment by lisitsyn** [comment]
Looks good to merge. Thanks!

**Pull request merged**